site/docs/tools/iceberg/flink.md

```diff
@@ -0,0 +1,94 @@
+
+In order to use Flink with Python API, you will need to make sure `pyflink` have access to all Hadoop JARs as mentioned in this [docs](https://iceberg.apache.org/flink/#preparation-when-using-flinks-python-api). After that, you will need to make sure `iceberg-flink-runtime` is added to Flink. This can be done by adding the iceberg JAR to `pyflink` via `env.add_jar`, e.g: `env.add_jars("file://path/to/jar/iceberg-flink-runtime-0.xx.x.jar")`. This can be shown below:
+
+```python
+import os
+
+from pyflink.datastream import StreamExecutionEnvironment
+
+env = StreamExecutionEnvironment.get_execution_environment()
+iceberg_flink_runtime_jar = os.path.join(os.getcwd(), "iceberg-flink-runtime-0.xx.x.jar")
+
+env.add_jars("file://{}".format(iceberg_flink_runtime_jar))
+```
+
+Once we have added `iceberg-flink-runtime` JAR to `pyflink`, we can then create `StreamTableEnvironment` and execute Flink SQL statements. This can be shown in the following example:
+
+```python
+from pyflink.table import StreamTableEnvironment
+
+table_env = StreamTableEnvironment.create(env)
+
+table_env.execute_sql(
+        """CREATE CATALOG <catalog_name> WITH (
+        'type'='iceberg',
+        'catalog-impl'='org.apache.iceberg.nessie.NessieCatalog',
+        'uri'='http://localhost:19120/api/v1',
+        'ref'='main',
+        'warehouse' = '/path/to/flink/warehouse')"""
+    )
+```
+
+With the above statement, we have created a Nessie catalog (via Iceberg) that Flink will use to manage the tables.
+
+For more general information about Flink and Iceberg, refer to [Iceberg and Flink documentation](https://iceberg.apache.org/flink/).
+
+
+## Configuration 
+
+To use Nessie Catalog in Flink via Iceberg, we will need to create a catalog in Flink through `CREATE CATALOG` SQL statement (replace `<catalog_name>` with the name of your catalog), example:
+
+```python
+table_env.execute_sql(
+        """CREATE CATALOG <catalog_name> WITH (
+        'type'='iceberg',
+        'catalog-impl'='org.apache.iceberg.nessie.NessieCatalog',
+        'uri'='http://localhost:19120/api/v1',
+        'ref'='main',
+        'warehouse' = '/path/to/flink/warehouse')"""
+    )
+```
+
+The following properties are **required** in Flink when creating the Nessie Catalog:
+
+- `type`: This **must** be `iceberg` for iceberg table format.
+- `catalog-impl`: This **must** be `org.apache.iceberg.nessie.NessieCatalog` in order to tell Flink to use Nessie catalog implementation.
+- `uri`: The location of the Nessie server.
+- `ref`: The Nessie ref/branch we want to use.
+- `warehouse`: The location where to store Iceberg tables managed by Nessie catalog.
+- `authentication.type`: The authentication type to be used, please refer to the [authentication docs](../auth_config.md) for more info.
```

Comment:
I think, technically `authentication.type` defaults to `NONE`, so it is not strictly required, but it'd worth mentioning anyway,